### PR TITLE
Fixes an issue with trailing spaces on directory path for Windows.

### PR DESCRIPTION
**Pull request by samds**
Hey guys,

On Windows, trailing spaces are automatically removed from directory name.
If I create a directory like that
`var dirInfo = System.IO.Directory.CreateDirectory(@"C:\User\Folder ");`
Then
`dirInfo.Name == "Folder"`
should be true.

### CODE REVIEW

File: System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryInfoTests.cs [hunk 1]
```diff
@@ -254,6 +254,38 @@ public void MockDirectoryInfo_FullName_ShouldReturnNormalizedPath(string directo
             Assert.AreEqual(expectedFullName, actualFullName);
         }
 
+        [TestCase(@"c:\temp\folder  ", @"c:\temp\folder")]
+        [WindowsOnly(WindowsSpecifics.Drives)]
+        public void MockDirectoryInfo_FullName_ShouldReturnPathWithTrimmedTrailingSpaces(string directoryPath, string expectedFullName)
+        {
+            // Arrange
+            directoryPath = XFS.Path(directoryPath);
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryInfoTests.cs]: No need for XFS.Path when the test is windows-specific

File: System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryInfoTests.cs [hunk 2]
```diff
@@ -254,6 +254,38 @@ public void MockDirectoryInfo_FullName_ShouldReturnNormalizedPath(string directo
...
+            // Assert
+            Assert.AreEqual(expectedFullName, actualFullName);
+        }
+
+        [TestCase(@"c:\temp\\folder ", @"folder")]
+        [WindowsOnly(WindowsSpecifics.Drives)]
+        public void MockDirectoryInfo_Name_ShouldReturnNameWithTrimmedTrailingSpaces(string directoryPath, string expectedName)
+        {
+            // Arrange
+            directoryPath = XFS.Path(directoryPath);
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryInfoTests.cs]: Same as above

Comment (samds) [reply System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryInfoTests.cs]: Thank you. The changes has been committed.

File: System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs [hunk 3]
```diff
@@ -24,7 +26,7 @@ public MockDirectoryInfo(IMockFileDataAccessor mockFileDataAccessor, string dire
             originalPath = directoryPath;
             directoryPath = mockFileDataAccessor.Path.GetFullPath(directoryPath);
 
-            this.directoryPath = directoryPath.TrimSlashes();
+            this.directoryPath = XFS.IsWindowsPlatform() ? directoryPath.TrimSlashes().TrimEnd(' ') : directoryPath.TrimSlashes();
```
Comment (304NotModified) [inline System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs]: same for here for `directoryPath.TrimSlashes()`

File: System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryInfoTests.cs [hunk 4]
```diff
@@ -254,6 +254,36 @@ public void MockDirectoryInfo_FullName_ShouldReturnNormalizedPath(string directo
             Assert.AreEqual(expectedFullName, actualFullName);
         }
 
+        [TestCase(@"c:\temp\folder  ", @"c:\temp\folder")]
+        [WindowsOnly(WindowsSpecifics.Drives)]
+        public void MockDirectoryInfo_FullName_ShouldReturnPathWithTrimmedTrailingSpaces(string directoryPath, string expectedFullName)
+        {
+            // Arrange
+            var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>());
```
Comment (304NotModified) [inline System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryInfoTests.cs]: FYI,

    var fileSystem = new MockFileSystem();

It also fine? (less = more)

2x

Comment (samds) [reply System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryInfoTests.cs]: Thanks. Done!

**Comment by fgreinacher** [comment]
Thanks!

**Pull request merged**